### PR TITLE
Update AWS Lambda SDK to 1.28.0

### PR DESCRIPTION
**Pull request by ohookins**
I've been scratching my head the last few days about this. The daemon itself works standalone, and will even do so packaged as a Docker image and used with lambda, but will run immediately and exit - not suitable for lambda usage. If you build the `lambda` subdirectory directly and produce a binary (presumably what is provided as `handler.zip` in release) it seems to work when packaged as a zip file but not when packaged as a Docker image.

I couldn't seem to determine what was causing problems but it's clearly:
- the current codebase
- when packaged as a Docker image and used as an AWS lambda function
- the binary itself and the code (e.g. `main()`) is executed, but the `Handler` function is never called by the AWS lambda runtime.

After trying a few different things, it seems like updating to a later version of the AWS Lambda SDK solves the issue. I suspect some subtle interoperation between the runtime and the specific version of this SDK is not working - after all, it's almost 4 years old at this point.

### CODE REVIEW

**Comment by ohookins** [comment]
For reference, this is how I'm building the Docker image:
```
FROM public.ecr.aws/lambda/provided:al2 as build

# See https://github.com/buildkite/buildkite-agent-metrics/releases for latest version
ENV LAMBDA_VERSION="5.3.0"

WORKDIR /app
COPY . /app

RUN yum install -y wget tar golang && \
    wget "https://github.com/buildkite/buildkite-agent-metrics/archive/refs/tags/v${LAMBDA_VERSION}.tar.gz" && \
    tar xvzf "v${LAMBDA_VERSION}.tar.gz" && \
    cd "buildkite-agent-metrics-${LAMBDA_VERSION}" && \
    go get ./lambda && \
    CGO_ENABLED=0 go build -o ../handler ./lambda

FROM public.ecr.aws/lambda/provided:al2

COPY --from=build /app/handler ${LAMBDA_TASK_ROOT}

ENTRYPOINT ["./handler"]
```

**Comment by moskyb** [comment]
hi there @ohookins! thanks for contributing.

this looks awesome! great work on the bug hunt - i suspect that the previous version of the lambda-go package we were using predates container lambdas, which i imagine is why it doesn't play nicely with them 😅

i'm happy to merge this as-is, but it seems like the dockerfile you've posted above might be useful to people as well - would you be interested in adding it as `lambda.Dockerfile` as part of this PR?

**Review by moskyb — APPROVED** [review]

**Comment by ohookins** [comment]
Yeah, I can add that no problem. I tend to use things like `Dockerfile.test` so perhaps `Dockerfile.lambda` ?

Actually there is a small problem with it I realise, AWS's provided:al2 ships with Golang 1.15 so in my local environment I've been using `golang:1.17` as the build image. I'll just retest the instructions and then add it to the branch if that still sounds ok.

**Comment by ohookins** [comment]
@moskyb anything else you need from me on this? How does that Dockerfile look? What's the release process from here?

File: Dockerfile.lambda [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+ENV LAMBDA_VERSION="5.3.0" 
+
+WORKDIR /app
+
+RUN apt-get update && apt-get install -y wget tar && \
+    wget "https://github.com/buildkite/buildkite-agent-metrics/archive/refs/tags/v${LAMBDA_VERSION}.tar.gz" && \
+    tar xvzf "v${LAMBDA_VERSION}.tar.gz" && \
+    cd "buildkite-agent-metrics-${LAMBDA_VERSION}" && \
+    go get ./lambda && \
+    CGO_ENABLED=0 go build -o ../handler ./lambda
```
Comment (moskyb) [inline Dockerfile.lambda:13]: we might be able to change this to use 
```
wget https://github.com/buildkite/buildkite-agent-metrics/releases/latest/download/handler.zip
```

and avoid actually having to build the binary as part of the docker build process, as well as removing the need for a static `LAMBDA_VERSION` envar

**Review by moskyb — APPROVED** [review]
i have a couple of comments on the dockerfile, but i'm also happy to merge this as is and make the changes myself - i'm aware of how annoying it might be for someone from some other company to say "yeah include that other thing, that looks great!" and then pick it to pieces at the next turn 😅

**Comment by moskyb** [comment]
re: the release process, once i have your okay, i'll merge it and make a release against this repo, which will publish the new versions of the handlers, binaries etc.

File: Dockerfile.lambda [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+FROM public.ecr.aws/lambda/provided:al2
+
+COPY --from=build /app/handler ${LAMBDA_TASK_ROOT}
```
[the remark this answers is not in the document]
Comment (ohookins) [reply Dockerfile.lambda:17]: It's a built-in variable in the provided AWS images: https://docs.aws.amazon.com/lambda/latest/dg/images-create.html#images-create-from-base

File: Dockerfile.lambda [hunk 3]
```diff
@@ -0,0 +1,19 @@
+FROM golang:1.17 AS build
```
[the remark this answers is not in the document]
Comment (ohookins) [reply Dockerfile.lambda:1]:
```suggestion
FROM golang:1.16 AS build
```

**Comment by ohookins** [comment]
You are free to merge it at any time! Up to you whether the LAMBDA_VERSION is a blocker or a nit.

File: Dockerfile.lambda (hunk 1 above)
Comment (ohookins) [reply Dockerfile.lambda:13]: I can fix it up to do this.

Comment (moskyb) [reply Dockerfile.lambda:13]: sounds great :)

**Comment by ohookins** [comment]
OK should be good to go now.

**Comment by moskyb** [comment]
Awesome! Thanks so much for your contribution @ohookins, i'm really happy with where we got to :)

**Pull request merged**